### ISSUE TITLE
Include symbols by default

### ISSUE THREAD

**Issue by loic-sharma** [NuGet/Home#7892]
The [snupkg spec](https://github.com/NuGet/Home/wiki/NuGet-Package-Debugging-&-Symbols-Improvements#solution-) mentions:

> The default NuGet Client tooling will automatically generate and publish Symbols packages when creating NuGet packages.

Currently, symbols are not generated by default. We should consider making this the default experience. See also: https://github.com/NuGet/Home/issues/7815

/cc @rrelyea @karann-msft @anangaur

**Comment by nkolev92** [comment]
I don't think symbols should be generated by default. 

I'd say, we need at least > 50% of the pack runs to be with symbols generated. 

Now we don't have the data for that, but can we comfortably say that 50% of all pack runs need symbols? 
I'd say symbols are only needed by default when you are going to publish the build output on a feed.

In my opinion, this would just add unnecessary overhead, and I don't see it as the day-to-day dev scenario. Especially when considering changing a long-standing default.

**Comment by anangaur** [comment]
I can argue both ways 😐. Let keep this open and see if it garners interest. We can come back to this once we have the community showing interest in it.

**Comment by loic-sharma** [comment]
> I'd say, we need at least > 50% of the pack runs to be with symbols generated.
> 
> Now we don't have the data for that, but can we comfortably say that 50% of all pack runs need symbols?
> I'd say symbols are only needed by default when you are going to publish the build output on a feed.

I'm curious about this, when would someone pack a project but not publish to a feed?

> I don't see it as the day-to-day dev scenario. Especially when considering changing a long-standing default.

The fact of the matter is, the vast majority of packages on nuget.org are not debuggable. This is largely because an author needs to opt-in to generate symbols. Could you imagine the benefits if instead authors had to *opt-out*? The .NET ecosystem would be dramatically better if developers could debug all of their dependencies.

**Comment by loic-sharma** [comment]
> I'm curious about this, when would someone pack a project but not publish to a feed?

Ah I forgot about `PackOnBuild`... Is this commonly used?

**Comment by nkolev92** [comment]
We don't really have data on all those metrics unfortunately which our decision more difficult. 

Take the dotnet repos (and the NuGet.Client repo itself). 
Take PR/private builds for example. All of those pack. Most of those don't generate or archive symbols. 

> The fact of the matter is, the vast majority of packages on nuget.org are not debuggable. 

I understand that and I agree with you. But, which packages don't have symbols is way more important than whether packages have symbols. 
Metrics that'd more telling in my opinion are, how many of the most popular 1000 packages pushed in the last year have symbols? It's new submissions that count. 

 > This is largely because an author needs to opt-in to generate symbols. 

There could be so many other reasons. For starters lack of proper tooling to create/lack of proper infrastructure to submit symbols. Break-down in the general publishing process. 

I don't believe strong-arming people with pack is the right decision. 

Defaults are as much about what the user expects, as they are about some ideal case scenario. 

Note that I'm not saying we should never do it. I'm just saying that the way everything is set-up right now, it's not the best direction.

**Comment by loic-sharma** [comment]
> Note that I'm not saying we should never do it. I'm just saying that the way everything is set-up right now, it's not the best direction.

Yeah, that's the sense I'm getting too now.

> For starters lack of proper tooling to create/lack of proper infrastructure to submit symbols. Break-down in the general publishing process.

I think this would help. Should Visual Studio's `Package` pane for project properties have an `Include symbols` option?

**Comment by nkolev92** [comment]
> I think this would help. Should Visual Studio's Package pane for project properties have an Include symbols option?

Maybe. The trouble I'm having is getting a representative sample of what people do & how they do it to make an educated decision. 
If people use the UI to model their packages, I agree with you.

We really need a "best practices" guidance from the team since we feel strongly about these topics.

**Comment by anangaur** [comment]
> We really need a "best practices" guidance from the team since we feel strongly about these topics.

Yup. We are indeed planning to make it a feature too, not just docs 🙂

**Comment by nkolev92** [comment]
`nuget <verb> -bestPractice` 😆
